public/app/features/search/page/table/Table.tsx

```diff
@@ -0,0 +1,200 @@
+    if (SEARCH_COLUMNS.includes(getFieldDisplayName(field, data))) {
+      const fieldTableOptions = (field.config.custom || {}) as TableFieldOptions;
+      const Cell = getCellComponent(fieldTableOptions.displayMode, field);
+      columns.push({
+        Cell: Cell,
+        id: fieldIndex.toString(),
+        field: field,
+        Header: getFieldDisplayName(field, data),
+        accessor: (row: any, i: number) => {
+          return field.values.get(i);
+        },
+        width: fieldTableOptions.width,
+      });
+    }
+  }
+
+  return columns;
+};
+
+export const Table = ({ data, width }: Props) => {
+  const styles = useStyles2(getStyles);
+  const tableStyles = useStyles2(getTableStyles);
+
+  const memoizedData = useMemo(() => {
+    if (!data.fields.length) {
+      return [];
+    }
+    // as we only use this to fake the length of our data set for react-table we need to make sure we always return an array
+    // filled with values at each index otherwise we'll end up trying to call accessRow for null|undefined value in
+    // https://github.com/tannerlinsley/react-table/blob/7be2fc9d8b5e223fc998af88865ae86a88792fdb/src/hooks/useTable.js#L585
+    return Array(data.length).fill(0);
+  }, [data]);
+
+  // React-table column definitions
+  const memoizedColumns = useMemo(() => generateColumns(data, width), [data, width]);
+
+  const options: TableOptions<{}> = useMemo(
+    () => ({
+      columns: memoizedColumns,
+      data: memoizedData,
+    }),
+    [memoizedColumns, memoizedData]
+  );
+
+  const { getTableProps, getTableBodyProps, headerGroups, rows, prepareRow } = useTable(options, useBlockLayout);
+
+  const RenderRow = React.useCallback(
+    ({ index: rowIndex, style }) => {
+      const row = rows[rowIndex];
+      prepareRow(row);
+
+      let nameField: Field<any, Vector<any>> | undefined = undefined;
+      let kindField: Field<any, Vector<any>> | undefined = undefined;
+      let urlField: Field<any, Vector<any>> | undefined = undefined;
+      for (const [_, field] of data.fields.entries()) {
+        const fieldDisplayName = field.name;
+        if (SEARCH_COLUMNS.includes(fieldDisplayName)) {
+          nameField = field;
+        }
+
+        if (fieldDisplayName === 'Kind') {
+          kindField = field;
+        }
+
+        if (fieldDisplayName === 'URL') {
+          urlField = field;
+        }
+
+        if (nameField && kindField && urlField) {
+          break;
+        }
+      }
+
+      return (
+        <div {...row.getRowProps({ style })} className={tableStyles.row}>
+          {row.cells.map((cell: Cell, index: number) => {
+            const valueIndex = nameField?.values.toArray().findIndex((value: Vector<any>, index) => {
+              if (value === cell.value) {
+                return index;
+              }
+              return;
+            });
+
+            const kind = kindField?.values.get(valueIndex!);
+            const url = urlField?.values.get(valueIndex!);
```

Comment:
Don't love generating this metadata for every single cell. Ideally there should be a more straightforward way of associating this data during initialization